Source/WarpX.cpp

```diff
@@ -254,6 +255,9 @@ WarpX::WarpX ()
     }
     do_back_transformed_particles = mypc->doBackTransformedDiagnostics();
 
+    // Particle Buffers
+    m_particle_buffers = std::make_unique<ParticleBoundaryBuffer>();
```

Comment:
is this par ticle buffer specific to boundaries?
In that case, should we name this to specifically be `m_particle_buffers_boundary`?

Reply:
I think at least currently it is used that way, so I'll push a renaming to make it clear (also we don't have a doxygen string on the member declaration yet to clarify)

Reply:
I would also make it singular, since this unique_ptr can only hold one instance of `ParticleBoundaryBuffer`

Reply:
@roelof-groenewald @PhilMiller please be aware of this renaming for your follow-up PR